app/src/main/java/org/simple/clinic/summary/updatephone/UpdatePhoneNumberDialog.kt

```diff
@@ -94,18 +112,25 @@ class UpdatePhoneNumberDialog : AppCompatDialogFragment(), UpdatePhoneNumberDial
     numberEditText.showKeyboard()
   }
 
+  @SuppressLint("CheckResult")
+  override fun onResume() {
+    super.onResume()
+    events
+        .takeUntil(screenDestroys)
+        .subscribe(dialogEvents::onNext)
```

Comment:
Alright, once we finish the migration, let's take a look at how dialogs work with our Mobius delegate and come up with something that works for all our cases.